public/offline.html

```diff
@@ -174,38 +187,74 @@ <h4>(pick a color below && start drawing!)</h4>
     </g>
   </svg>
   <script type="text/javascript">
+
+    const palettes = {
+      default: ["#F4908E", "#F2F097", "#88B0DC", "#F7B5D1", "#53C4AF", "#FDE38C"],
+      // Color-deficient accessible palettes: https://owi.usgs.gov/blog/tolcolors/
+      paulTol: ["#B997C6", "#824D99", "#4E79C4", "#57A2AC", "#7EB875", "#D0B440", "#E67F33"],
+      // Add more palette options below and to the "select" dropdown.
+    }
+    let activePalette = 'default' // match key inside the palettes variable
+
+    // Globally shared DOM nodes
     const canvas = document.querySelector('canvas')
     const context = canvas.getContext('2d')
-    const colors = ["#F4908E", "#F2F097", "#88B0DC", "#F7B5D1", "#53C4AF", "#FDE38C"]
     const colorDiv = document.querySelector(".colors")
 
-    colors.forEach(color => {
-      const button = document.createElement("button")
-      button.classList.add("color")
-      button.style.backgroundColor = color
-      colorDiv.appendChild(button)
-      button.addEventListener('click', () => context.strokeStyle = color)
-    })
+    // handler for color input buttons
+    function handleButtonClick(event) {
+      const buttonStyle = event.target.style
+      context.strokeStyle = buttonStyle.backgroundColor
+    }
+
+    // Main drawing function
+    function renderColorButtons(colors) {
+      // First, remove existing event listeners to prevent memory leak
+      colorDiv.querySelectorAll('button').forEach((button) => {
+        button.removeEventListener('click', handleButtonClick)
+      })
+      // Then, remove existing buttons
+      colorDiv.innerHTML = ""
+
+      // Lastly, add new a new button for each color in the passed in palette
+      colors.forEach(color => {
+        const button = document.createElement("button")
+        button.classList.add("color")
+        button.style.backgroundColor = color
+        colorDiv.appendChild(button)
+        button.addEventListener('click', handleButtonClick)
+      })
+    }
+
+    // Helper for modifying canvas context
+    const setContextSettings = (strokeColor) => {
```

Comment:
I separated this section out from `setSize` since these lines were unrelated to changing the size of the canvas.